posts/2024-03-21-Rust-1.77.0.md

```diff
@@ -0,0 +1,124 @@
+---
+layout: post
+title: "Announcing Rust 1.77.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.77.0. Rust is a programming language empowering everyone to build reliable and efficient software.
+
+If you have a previous version of Rust installed via `rustup`, you can get 1.77.0 with:
+
+```console
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.77.0](https://doc.rust-lang.org/nightly/releases.html#version-77-2024-03-21).
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.77.0 stable
+
+This release is relatively minor, but as always, even incremental improvements lead to a greater whole. A few of those changes are highlighted in this post, and others may yet fill more niche needs.
+
+### C-string literals
+
+Rust now supports C-string literals (`c"abc"`) which expand to a nul-byte
+terminated string in memory of type `&'static CStr`. This makes it easier to write code
+interoperating with foreign language interfaces which require nul-terminated
+strings, with all of the relevant error checking (e.g., lack of interior nul
+byte) performed at compile time.
+
+### Support for recursion in `async fn`
+
+async functions previously could not call themselves due to a compiler analysis
+limitation. In 1.77, that limitation has been lifted, so recursive calls are
```

Comment:
```suggestion
Async functions previously could not call themselves due to a compiler
limitation. In 1.77, that limitation has been lifted, so recursive calls are
```
the limitation was not in our analysis, it was in our internal representation of the type of the state machine. The state machines were structural types, so any recursive async function would have previously resulted in a recursive type. It has the same underlying cause preventing tuples from containing references to itself, while tuple structs are allowed to do so.